lib/phil_web/live/charlie_cards/importer.ex

```diff
@@ -0,0 +1,114 @@
+defmodule PhilWeb.Live.CharlieCards.Importer do
+  @moduledoc """
+  LiveView for importing CharlieCard inventory into Phil.Application
+
+  An external module `Phil.CharlieCards.Importer` handles all of the importing logic and returns a map of successes and failures.
+
+  This LiveView handles the upload of the file and reports on successes and failures.
+  """
+
+  use PhilWeb, :live_view
+
+  alias Phil.CharlieCards.Importer
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div>
+      <.upload_form uploads={@uploads} />
+      <.results results={@results} />
+    </div>
+    """
+  end
+
+  @impl true
+  def handle_event("process", _params, socket) do
+    results =
+      consume_uploaded_entries(socket, :inventory, fn %{path: path}, _ ->
+        {:ok, Importer.import(path)}
+      end)
+      |> List.first()
+
+    {:noreply, assign(socket, results: Map.merge(socket.assigns.results, results))}
```

Comment:
I don't think you need to merge with existing as the merge will alway overwrite it anwyway

Reply:
You do because the results won't always come back with both an :error or :ok key. For example, they could all be :ok. If you don't do this, you have to check if the key exists before checking the length of the entries.

Reply:
No, it will still work because you reset to %{error: [], ok: []} and then just merge the results into that every time.

Reply:
I don't understand this at all. If `results` can _not_ have a key in it, then surely this is possible? What do you mean reset? when?

```elixir
assigns = %{:ok: [], error: [1]}
results = %{ok: [1]}
Map.merge(assigns, results)
```

Reply:
I just changed it so that the results always come back with both keys.